### DISCUSSION TITLE
Gitstream does not merge when using merge queues

### DISCUSSION THREAD

**Discussion by MrGreenTea** [linear-b/gitstream#484] [Troubleshooting]
We're using gitstream in our repos and recently started using merge queues, especially in combination with dependabot.
Before we used merge queue gitstream used to auto-merge (most of the time) dependabot PRs as configured.

Since the move to merge queue it does not work anymore and we have to manually add the PRs to merge queue.

See for example this PR: https://github.com/YoungVision-eV/website/pull/138

I haven't found much in the documentation besides [this small answer](https://docs.gitstream.cm/faq/#can-i-use-gitstream-with-merge-queues) but that's not for auto merging but just for required checks.

What can/should we do to get back our auto merges of dependabot updates?

**Comment by PavelLinearB** [comment]
Hi @MrGreenTea,
I am looking into this. Do you get an error in the gitStream checks on GitHub? 
If so, can you please share the message here, along with a screenshot of the concluded checks?

Thanks

**Reply by MrGreenTea** [reply]
Yes, it says this:
> approve@v1: approve the PR
> add-label@v1: add label approved-dependabot
> merge@v1: Waiting for required checks and approvals, details: Changes must be made through the merge queue

**Reply by PavelLinearB** [reply]
I see
I suspect this is because the `merge` API differs from the one that adds the PR to the merge queue.
We will work to support this functionality. I will update here with an ETA

**Reply by MrGreenTea** [reply]
Yeah I found the [enqueuePullRequest](https://docs.github.com/en/graphql/reference/mutations#enqueuepullrequest) mutation for this
